### PR TITLE
Restore python_shell for onlyif/unless

### LINKED ISSUE #20936: critical: develop: onlyif/unless is broken (python_shell=False)
This for evident compatiblity amongst existant states tree which have elaborated shell tests for onlyif/unless.
I think it should by on by default, or at least togglable on or off.
So, what to do ? another config opt or a blind python_shell=True

cc @thatch45 @rallytime

### PR DESCRIPTION
**Pull request by kiorky**
This fixes #20936.

### CODE REVIEW

**Comment by kiorky** [comment]
ha. i will have a look to failures :)

**Comment by kiorky** [comment]
so the failures are related, just a typo, i will add the fixes, :)

**Comment by kiorky** [comment]
lint unrelated

File: salt/modules/zcbuildout.py [hunk 1]
```diff
@@ -1032,14 +1032,20 @@ def _check_onlyif_unless(onlyif, unless, directory, runas=None, env=()):
                 if not onlyif:
                     _valid(status, 'onlyif execution failed')
             elif isinstance(onlyif, string_types):
-                if retcode(onlyif, cwd=directory, runas=runas, env=env) != 0:
+                if retcode(
```
Comment (thatch45) [inline salt/modules/zcbuildout.py]: Can we separate the zcbuildout update here into a separate PR since it functions in the module domain instead of the state domain? Making shell changes in modules introduces a separate security context then in the state system.

Comment (kiorky) [inline salt/modules/zcbuildout.py]: np, but the default value isnt changed after all if you reread well, im going to split into two and rebase this request to include only one change (state i think) and open the other counter part.

Comment (kiorky) [inline salt/modules/zcbuildout.py]: whoops, i thought that was the _Popen add, sry misread. let me split the pr

Comment (kiorky) [inline salt/modules/zcbuildout.py]: done in #20991

**Comment by kiorky** [comment]
rels #20991

**Comment by thatch45** [comment]
Thanks @kiorky !
